src/LinuxTracing/UprobesUnwindingVisitor.cpp

```diff
@@ -528,22 +528,44 @@ void UprobesUnwindingVisitor::Visit(uint64_t event_timestamp,
 
 void UprobesUnwindingVisitor::Visit(uint64_t event_timestamp,
                                     const UserSpaceFunctionEntryPerfEventData& event_data) {
-  function_call_manager_->ProcessFunctionEntry(event_data.tid, event_data.function_id,
-                                               event_timestamp, std::nullopt);
+  if (!tid_to_root_namespace_tid_.contains(event_data.tid)) {
```

Comment:
It's hard to say for me, but I think here we should avoid the double lookup in the map and use `find` instead and check for `end()`.